### ISSUE TITLE
Cursor style changes to block and then it is persistent

### ISSUE THREAD

**Issue by j605** [wsdjeg/SpaceVim#318]
## Expected behavior, english is recommend
The cursor style if modified in spacevim shouldn't affect the terminal cursor style after quitting it.

## Environment Information
- OS: ArchLinux
- vim version: NA
- neovim version: 0.1.7

## The reproduce ways from Vim starting (Required!)
Start neovim in st and quit it. The cursor style is change and cannot be reverted. In tmux it is persistent for the whole session.

## Output of the ":message" command, and ":echo SpaceVim#logger#viewLog()"
No output for :message. 
Output for the second command,
### SpaceVim Options :

```viml
g:spacevim_enable_os_fileformat_icon = 0
g:spacevim_version = '0.2.0-dev'
g:spacevim_force_global_config = 0
g:
    spacevim_smartcloseignoreft = ['help']
g:spacevim_default_indent = 2
g:spacevim_enable_javacomplete2_py = 0
g:spacevim_enable_vimfiler_welcome = 1
g:
    spacevim_colorscheme_default = 'desert'
g:spacevim_warning_symbol = '⚠'
g:spacevim_simple_mode = 0
g:spacevim_colorscheme = 'gruvbox'
g:spacevim_enable_guicolors
    = 1
g:spacevim_custom_plugins = []
g:spacevim_hiddenfileinfo = 1
g:spacevim_error_symbol = '✖'
g:spacevim_vim_plug_installed = 0
g:spacevim_windows_leader =
    's'
g:spacevim_wildignore = '*/tmp/*,*.so,*.swp,*.zip,*.class,tags,*.jpg,*.ttf,*.TTF,*.png,*/target/*,.git,.svn,.hg,.DS_Store'
g:spacevim_use_colorscheme = 1
g:
    spacevim_plugin_manager = 'dein'
g:spacevim_enable_neocomplcache = 0
g:spacevim_smartcloseignorewin = ['__Tagbar__', 'vimfiler:default']
g:spacevim_enable_debug =
    0
g:spacevim_buffer_index_type = 0
g:spacevim_vim_help_language = 'en'
g:spacevim_lint_on_save = 1
g:spacevim_dein_installed = 1
g:spacevim_autocomplete_method
    = 'deoplete'
g:spacevim_unite_leader = 'f'
g:spacevim_enable_tabline_filetype_icon = 0
g:spacevim_enable_powerline_fonts = 1
g:spacevim_plugin_groups_exclude =
    []
g:spacevim_neobundle_installed = 0
g:spacevim_disabled_plugins = []
g:spacevim_guifont = ''
g:spacevim_plugin_groups = ['web', 'lang', 'edit', 'ui', 'tools',
    'checkers', 'format', 'chat', 'git', 'javascript', 'ruby', 'python', 'scala', 'lang#go', 'lang#markdown', 'scm', 'editing', 'indents', 'navigation', 'misc', 'core',
    'unite', 'github', 'denite', 'ctrlp', 'autocomplete', 'nvim', 'linux', 'colorscheme']
g:spacevim_sidebar_width = 30
g:spacevim_enable_cursorline = 0
g:
    spacevim_enable_googlesuggest = 0
g:spacevim_language = ''
g:spacevim_enable_neomake = 1
g:spacevim_plugin_bundle_dir = '/home/jagan/.cache/vimfiles/'
g:
    spacevim_hosts_url = 'https://raw.githubusercontent.com/racaljk/hosts/master/hosts'
g:spacevim_altmoveignoreft = ['Tagbar', 'vimfiler']
g:
    spacevim_enable_cursorcolumn = 0
g:spacevim_realtime_leader_guide = 0
g:spacevim_enable_ycm = 0
g:spacevim_checkinstall = 1
g:spacevim_plugin_name = 'gruvbox'
g:
    spacevim_debug_level = 1
g:spacevim_filemanager = 'vimfiler'
g:spacevim_src_root = 'E:\sources\'
g:spacevim_max_column = 120
```

### SpaceVim Health
    checking :

SpaceVim clipboard support check report:
Checking +clipboard:
      SUCCEED!
SpaceVim lua support check report:
Checking +lua:
      Failed :
    Known issue, neovim do not support lua now.
SpaceVim python support check report:
Checking +python3:
      SUCCEED!
Checking +python:
      SUCCEED!

###
    SpaceVim runtime log :

```log
[ SpaceVim ] : logger file /home/jagan/.SpaceVim/.SpaceVim.log does not exists, only log for current process will be shown!
```

please post log below, if you want me reproduce your issue quickly, post your custom config here will be better.

**Comment by wsdjeg** [comment]
It is feature of neovim 0.1.7, Now I am using neovim 0.2.0-dev, and it works well for me.

**Comment by j605** [comment]
The bug is solved with `neovim 2.0` (which was released recently) like you suggested.
